ui/client/components/graph/node-modal/editors/expression/EditableExpression.js

```diff
@@ -0,0 +1,71 @@
+import RawEditor from "./RawEditor"
+import BoolEditor from "./BoolEditor"
+import React from "react"
+import ExpressionWithFixedValues from "./ExpressionWithFixedValues"
+import {parseableBoolean} from "./ExpressionParser"
+
+export default class EditableExpression extends React.Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      displayRawEditor: true
+    }
+  }
+
+  render() {
+    const {fieldType, expressionObj, rowClassName, valueClassName, showSwitch} = this.props
+    const editorName = resolveEditorName(fieldType, expressionObj, this.state.displayRawEditor)
+    const Editor = editorTypes[editorName]
+    return <Editor toggleEditor={this.toggleEditor}
+                   switchable={this.switchable(editorName, expressionObj)}
+                   shouldShowSwitch={this.showSwitch(fieldType, showSwitch)}
+                   rowClassName={rowClassName ? rowClassName : "node-row"}
+                   valueClassName={valueClassName ? valueClassName : "node-value"}
+                   {...this.props}/>
+  }
+
+  toggleEditor = (_) => this.setState({
+    displayRawEditor: !this.state.displayRawEditor
+  })
+
+  switchable = (editorName, expressionObj) => {
+    switch (editorName) {
+      case boolEditor:
+        return true
+      case rawEditor:
+        return parseableBoolean(expressionObj)
```

Comment:
It's use only EditableExpression.js move it to this file ;)

Reply:
Yes, but I think separation of responsibilities should be considered here - there will be more parse functions in the near future. I don't want to keep all these functions in React.Component